lib/twilio/audiohelper.ts

```diff
@@ -90,6 +90,11 @@ class AudioHelper extends EventEmitter {
    */
   private _audioContext?: AudioContext;
 
+  /**
+   * The enumerateDevices method to use
+   */
+  private _enumerateDevices: any;
```

Comment:
This has to be `any` since we're allowing the customers to provide any custom type. If we make this strict e.g. same type as `mediaDevices.enumerateDevices`, customers will have to update the type of their override.

Reply:
I'm thinking that might be better, no? If they use it like this
```ts
const dev = new Device(token, { enumerateDevices: myEnumDevicesShim as any })
```
This feels clearer in the sense that it's up to them to provide a proper shim, and if the shim they're using isn't up to spec this type-cast makes it clear they're taking that responsibility.

Reply:
The concern is, the default type may contain a bunch of required members which are not necessarily required for the functionality needed. We're putting more work for the customer if we do this. They either need to do what you suggested above, or define all the required members, which are all unnecessary steps.